### ISSUE TITLE
[ANNOUNCEMENT] Gold is deprecated

### ISSUE THREAD

**Issue by yosssi** [yosssi/gold#24]
This is an announcement for all who use Gold.

Hi, I'm an author of Gold. Thanks for using this package.

I decided to re-create this package as a new one because I wanted to refine and simplify the source code, data structure and APIs of this package for its future enhancement.

I just finished creating a new HTML template engine and released it.

[Ace - HTML template engine for Go](https://github.com/yosssi/ace)

I would be glad if you could use this new package hereafter.

Best regards,
Keiji

**Comment by ahall** [comment]
I've been using Gold for a while and you implemented some of functionality I needed like #4, #10, #8, #7 and some more. Is this functionality in Ace yet? Also need sublime editor support.

**Comment by yosssi** [comment]
@ahall Thanks for your contributions to Gold! I studied a lot from you and I implemented some functions which you proposed in Ace.
- #4 Optional Blocks: Implemented in Ace. See https://github.com/yosssi/ace/blob/master/documentation/syntax.md#yield-helper-method
- #10 Template Base Directory: Implemented in Ace. See the "BaseDirectory" option in https://godoc.org/github.com/yosssi/ace#Options
- #7 Template Macros: Implemented in Ace. See https://github.com/yosssi/ace/tree/master/examples/pass_pipeline_to_included_template.
- #8 Debugging Errors: Not implemented in Ace. I'll think of implementing this function in Ace.

And I'll create a Sublime syntax highlighter at yosssi/ace#20

Thanks a lot!

**Comment by ahall** [comment]
Thanks @yosssi, excellent news!

**Comment by yosssi** [comment]
@ahall Created a Sublime syntax highlighter. Please check it out. https://github.com/yosssi/ace-tmbundle

**Comment by ahall** [comment]
Hi @yosssi and thanks for your great work. Have you had chance to look at #8 yet? I want to switch from gold to ace for my application as soon as possible.

**Comment by yosssi** [comment]
Hi @ahall, I'm so sorry for keeping you wait. Later, I'll check #8. Perhaps you can solve your problem by using [gohtml](https://github.com/yosssi/gohtml)

**Comment by ahall** [comment]
Hi @yosssi, been looking at Ace and got couple of questions.

Not sure about the syntax changes between gold and ace. I prefer the way you extend a base template in gold instead of providing it in code using base and inner. It's just less extensible means you can't have chain of extends. Why the change? Gold syntax was pretty nice in my opinion.

**Comment by ahall** [comment]
One more :). The following doesn't work like it did in gold:

```
= include helpers/render_input name="password" type="password" value="" placeholder="Password" icon="icon-lock" login_input="true" required="true"
```

render_input_name.ace is as follows:

```
{{$field := "%{name}"}}
{{$required := eq "%{required}" "true"}}
{{$loginInput := eq "%{login_input}" "true"}}
{{$type := "%{type}"}}
div class="form-group{{FormGroupExtra .State $field}}"
  div class="{{if not $loginInput}}col-sm-6{{end}}"
    label class="block clearfix"
      span class="block input-icon input-icon-right"
        {{$error := FieldError $field .State}}
        input name="{{$field}}" type="{{if $type}}{{$type}}{{else}}text{{end}}" class="form-control" placeholder="%{placeholder}" value="%{value}" [{{if $required}}required{{end}}]
        i class="%{icon}"
        {{if $error}}
        span class="help-block error" {{ $error }}
        {{end}}
```

Basically trying to move all our workaround from earlier form Gold to Ace. Cheers.

**Comment by yosssi** [comment]
Regarding #8, you can print the intermediate HTML by using [gohtml](https://github.com/yosssi/gohtml) Please see [this example](https://gist.github.com/yosssi/70ea023495531ea1a01b).

**Comment by yosssi** [comment]
@ahall 

> It's just less extensible means you can't have chain of extends.

You can't have chain of extends also by using Gold.

> Why the change?

I changed the extending syntax because the Gold style of specifying a parent template in a child template is not extensible. The ace style of detaching of a notation of a parent template from a child template is more extensible because we can reutilize the child template with other parent templates.

**Comment by yosssi** [comment]
@ahall Regarding including, please see [this example](https://github.com/yosssi/ace/tree/master/examples/pass_pipeline_to_included_template).

**Comment by ahall** [comment]
Yeah I had seen that include example, but it doesn't let me name the parameters in order to make a reusable function to render form inputs. I need exactly the same as we discussed in #7.

**Comment by yosssi** [comment]
@ahall You can pass parameters to the included template by using a custom function. Please get a new version of Ace(`go get -u github.com/yosssi/ace`) and try [this example](https://gist.github.com/yosssi/74cb7804724acb22b2ff).

**Comment by ahall** [comment]
Thanks @yosssi
